pkg/controllers/deprovisioning/controller.go

```diff
@@ -110,17 +110,20 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
+		logging.FromContext(ctx).Debugf("waiting for in-memory cluster state to sync before deprovisioning")
```

Comment:
This is going to be really, really spammy while it's waiting for cluster state to sync during a scale-up. I'd prefer a gauge metric or even a liveness check on this to ensure that we don't fail to sync the cluster state for an extended period of time

Reply:
+1 to keep this log for now until any issues with deprovisioning are resolved.

Reply:
This just seems like the wrong mechanism. A gauge metric would give us historic data and give us the same info

Reply:
Except for cases when customers aren't already collecting those metrics. I'm interested in quickly solving the current burning issue that appears to indicate deprovisioning is in some way hanging.

Reply:
I think out of all the log lines, I think this one should stay. I'm iffy on the others, because for a stable running cluster, it's (# of deprovisioners) log lines every 10 seconds. This one should only happen in a cluster with a lot of churn or a wedged cluster state, which I argue is a reasonable thing.

Reply:
Agree with Nick on this.